service/handlers_test.go

```diff
@@ -842,7 +842,11 @@ func TestCreateWorkflowFromGit(t *testing.T) {
 				},
 			},
 			wfMock: &th.WorkflowMock{
-				SubmitFunc: func(ctx context.Context, from string, parameters, labels map[string]string) (string, error) {
+				SubmitFunc: func(ctx context.Context, from string, parameters map[string]string, labels map[string]string) (string, error) {
+					if parameters["environment_variables_string"] != "env foobar='barfoo'" {
```

Comment:
Lets add in more tests of that exercise different values in the environment variables. There are many combinations here that could trip up the quote replacement logic.